src/aggregate.js

```diff
@@ -34,29 +35,28 @@ const config = require('./config');
  *    -after {String} The _id to start querying the page.
  *    -before {String} The _id to start querying previous page.
  *    -options {Object} Aggregation options
+ * @param {(() => number) | undefined | null} getPipelineIndexFn Functor that determines where to insert the pagination-specific operations in the pipeline
```

Comment:
Please add a unit test demonstrating a "real world" scenario for this parameter. My understanding from reading the code is that it's useful only if you have multiple `$match` stages in your aggregation. This parameter will allow you to pick where in the pipeline to insert the 3 stages required for pagination.

Please add unit tests that cover the newly added lines. We want to maintain near-100% unit test coverage in this module.